v3/gssapi/client.go

```diff
@@ -212,3 +216,48 @@ func (client *Client) NegotiateSaslAuth(input []byte, authzid string) ([]byte, e
 
 	return output, nil
 }
+
+func UnmarshalWrapToken(wt *gssapi.WrapToken, data []byte, expectFromAcceptor bool) error {
+	// Check if we can read a whole header
+	if len(data) < 16 {
+		return errors.New("bytes shorter than header length")
+	}
+
+	// Is the Token ID correct?
+	expectedWrapTokenId := [2]byte{0x05, 0x04}
+	if !bytes.Equal(expectedWrapTokenId[:], data[0:2]) {
+		return fmt.Errorf("wrong Token ID. Expected %s, was %s", hex.EncodeToString(expectedWrapTokenId[:]), hex.EncodeToString(data[0:2]))
```

Comment:
In cases where returning additional context with the error it may be nice for it to wrap a simpler 'known' error as the related comment says.

eg `return fmt.Errorf("wrong blah blah. expected %s, was %s: %w", x, y, ErrUnmarshalFault)` or something like that